skll/data/readers.py

```diff
@@ -95,6 +94,7 @@ def __init__(self, path_or_list, quiet=True, ids_to_floats=False,
         self.id_col = id_col
         self.class_map = class_map
         self._progress_msg = ''
+        self._use_pandas = False
```

Comment:
Ah, so are we keeping the old way of reading in the files? What's the motivation for that?

Reply:
Good question, I should've explained this better above.

It makes sense to use `pandas` to read `TSV`, `CSV`, and `ARFF` files, since  either (1) `pandas` either already offers a reader for the file type (TSV, CSV), or (2) the data can easily be manipulated to be read with `pandas` (ARFF). However, that is not the case with `MegaM`, `NDJ`, `LibSVM`, and `DictList`.

The idea is that we'll keep the regular format (reading element-by-element) these file types that cannot be easily converted to be read with `pandas`. For `TSV`, `CSV`, and `ARFF`, we'll just read them directly into `pd.DataFrame` objects. The `_use_pandas` attribute just indicates whether the specific subclass uses a `pandas` reader, or continues doing things the old way.

Reply:
1. I was debating whether to use `pandas` for some of the `writer` objects. I never benchmarked whether writing would be faster with `pandas`. I'll do some testing and see whether it's worth it.

2. Yes, absolutely, I'll to that, too.

3. Yup, I think this makes sense. I'll update.

Reply:
Okay, so now the `use_pandas`  attribute makes sense. However, couldn't we handle NDJ/jsonlines files too using some combination of `apply()` and `pd.read_json()`?

Re: my engine comment, may be you were already thinking this but perhaps we should try and use the C engine and fall back to the Python engine if that fails.